### PR TITLE
Typescript 4.9 support for v1.1 code

### PR DESCRIPTION
**Pull request by mfedderly**
- Change examples/one_dot_one to use ts4.9
- Fix errors coming from api, generator, and legacy-client related to the old compiler

### CODE REVIEW

File: .monorepolint.config.mjs [hunk 1]
```diff
@@ -52,10 +52,28 @@ function standardPackageRules(shared) {
...
+          compilerOptions: {
+            rootDir: "src",
+            outDir: "build/types",
+            composite: true,
+          },
+          include: ["./src/**/*", ".eslintrc.cjs"],
+        },
+      },
+    }),
+    standardTsconfig({
```
Comment (mfedderly) [inline .monorepolint.config.mjs:70]: @ericanderson is there a way to make this less duplicitive?

Comment (ericanderson) [reply .monorepolint.config.mjs:70]: Either a function for reuse of the stnadardTsConfig. You can also give a generator function which passes you some details that might make it more programatic.

Comment (ericanderson) [reply .monorepolint.config.mjs:70]: Also this is so sad. The old includes were so much better!

File: examples/one_dot_one/package.json [hunk 2]
```diff
@@ -33,7 +33,7 @@
   },
   "devDependencies": {
     "@types/node": "^18.0.0",
-    "typescript": "^5.2.2"
+    "typescript": "^4.9.0"
```
Comment (ericanderson) [inline examples/one_dot_one/package.json:36]: We should probably use MRL to make sure we don't break the invariant of certain packages being 4x

**Review by ericanderson — APPROVED** [review]

**Pull request merged**